src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,176 @@
+        else:
+            print(f"Unexpected status code: {response.status_code}")
+            raise Exception("Unexpected status code")
+
+    # Parse the JSON response
+    try:
+        data = response.json()["daily"]
+    except (ValueError, KeyError) as e:
+        print(f"Error parsing API response: {e}")
+        raise
+    return data
+
+
+def model_data(city: str) -> pd.DataFrame:
+    data = {"date": years}
+
+    for model in MODELS:
+        results = call_api(city, model)
+        data[f"temperature_2m_mean_{model}"] = np.asarray(
+            results["temperature_2m_mean"]
+        )
+        data[f"precipitation_sum_{model}"] = np.asarray(results["precipitation_sum"])
+        data[f"soil_moisture_0_to_10cm_mean_{model}"] = np.asarray(
+            results["soil_moisture_0_to_10cm_mean"]
+        )
+
+    df = pd.DataFrame(data)
+
+    # Calculate mean
+    temp_columns = [col for col in df.columns if "temperature_2m_mean" in col]
+    precip_columns = [col for col in df.columns if "precipitation_sum" in col]
+    soil_moisture_columns = [
+        col for col in df.columns if "soil_moisture_0_to_10cm_mean" in col
+    ]
+
+    df["temperature_2m_mean"] = df[temp_columns].mean(axis=1)
+    df["precipitation_sum"] = df[precip_columns].mean(axis=1)
+    df["soil_moisture_0_to_10cm_mean"] = df[soil_moisture_columns].mean(axis=1)
+
+    # Select necessary columns
+    df = df[
+        [
+            "date",
+            "temperature_2m_mean",
+            "precipitation_sum",
+            "soil_moisture_0_to_10cm_mean",
+        ]
+    ]
+
+    return df
+
+
+def plot_climate_data(df: pd.DataFrame, city: str) -> plt:
+    numeric_df = df.select_dtypes(include=[np.number])
+    numeric_df = numeric_df.rolling(window=12).mean()
+
+    # Create a new figure with two subplots, sharing the x axis
+    fig, (ax1, ax3) = plt.subplots(2, 1, figsize=(10, 12), sharex=True)
+
+    # Create a color palette
+    tableau_palette = list(mcolors.TABLEAU_COLORS.values())
+
+    # First subplot
+    ax2 = ax1.twinx()
+    ax1.plot(
+        numeric_df.index,
+        numeric_df["temperature_2m_mean"],
+        color=tableau_palette[0],
+        label="Mean Temperature (ºC)",
+    )
+    ax2.plot(
+        numeric_df.index,
+        numeric_df["soil_moisture_0_to_10cm_mean"],
+        color=tableau_palette[1],
+        label="Mean Soil Moisture 0-10cm (m^3/m^3)",
+    )
+
+    # Set the y-axis labels
+    ax1.set_ylabel("Mean Temperature (ºC)", color=tableau_palette[0])
+    ax2.set_ylabel("Mean Soil Moisture 0-10cm (m^3/m^3)", color=tableau_palette[1])
+
+    # Set the x-axis major ticks to the decades.
+    years = mdates.YearLocator(10)
+    ax1.xaxis.set_major_locator(years)
+
+    # Add a legend
+    lines1, labels1 = ax1.get_legend_handles_labels()
+    lines2, labels2 = ax2.get_legend_handles_labels()
+    ax2.legend(lines1 + lines2, labels1 + labels2, loc=0)
+
+    # Second subplot
+    ax3.bar(
```

Comment:
I'm checking the resulting plot and it' s not great: all the bars look almost the same and there are not many insights you can get from it. I recommend a normal scatter + plot and I would change the limits of the y axis